### DISCUSSION TITLE
Hyperdeck Clip Playlist

### DISCUSSION THREAD

**Discussion by randallpacker** [danielbuechele/atemOSC#266] [Ideas]
Greetings: With the current functionality of atemOSC, is it possible to create Hyperdeck clip playlists? The challenge (for me) is knowing when a clip ends before starting the next. If it's possible to get a message from the endpoint of a clip, it would then be pretty straightforward to trigger the next clip: either as a linear or non-linear sequence. I use Max/MSP so I can easily create a sequence if I can get the timing information, either duration or endpoint. 

Perhaps it is possible to do this? 

Thanks! Randall

**Comment by randallpacker** [comment]
I have been going through the documentation it seems as though if you are polling 'state' and it changes from 'play' to 'idle' that would be the end of the clip and the trigger for the next clip. Not sure though how to set that up. Peter, can you provide an example script for grabbing the state? I understand it's a string but not sure how to script it. Thanks!!

**Comment by SteffeyDev** [comment]
The state should be sent out as feedback automatically. Are you seeing this in the log when the clip ends? If you see a state change on clip end, that should be sufficient for your purposes. As far as an example script, I have no clue how Max/MSP scripting works. Is it a visual script editor? JavaScript? Python?

**Comment by randallpacker** [comment]
Thanks Peter. I corrected the feedback messaging IP address, so now the state and clip messages are coming in to atemOSC, making it easy to then setup a receiver in Max. I haven't tested yet, but I'll be interested to see how seamlessly I can trigger a new clip after receiving the 'idle' message. I am hoping for clean clip changes without any pausing or glitch, as I plan to sequence the clips at a fairly fast rate, with durations as short as 1/4 second. I'll keep you posted.

**Comment by randallpacker** [comment]
Now that I have the feedback messaging working, I have discovered a couple of things I would like to bring to your attention. I have attached a screenshot of a Max patch for reference: 

1. If you look at the top of the patch where I have my sending commands, I am controlling Hyperdeck 1 with hyperdeck/1. However in the feedback messaging, Hyperdeck 1 is received as hyperdeck/0. So sending begins with 1 and receiving begins with 0. 
2. Timeline-time and Clip-time are identical. If you look at the Clip number (3) they should be different times?  However they are the same. 

If anyone is interested, I am would be happy to attach the original Max patch. It's great software for OSC. Max is object-oriented visual programming software for data, music, and video from Cycling '74: https://www.cycling74.com

[](url
<img width="790" alt="Screen Shot 2022-07-26 at 12 20 57 PM" src="https://user-images.githubusercontent.com/3989315/181059333-7769001c-ad16-4b19-ac8d-4a16957cbb7c.png">
)
